src/task.ts

```diff
@@ -104,6 +113,136 @@ export async function addProvingTask(
   console.log("Finish addProvingTask!");
 }
 
+function sleep(ms: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+
+async function sendIntervaledRequests(
+  tot_ms: number,
+  interval_ms: number,
+  n_req: number,
+  request_fn: (i : number) => Promise<void>
+) {
+    const start_t = Date.now();
+    let req_cnt = 0;
+
+    for (let curr_t = Date.now(); curr_t - start_t < tot_ms && req_cnt < n_req; curr_t = Date.now()) {
+        await request_fn(req_cnt);
+        req_cnt++;
+        await sleep(interval_ms);
+    }
+
+    console.log("Finished sending requests.");
+}
+
+async function runProveTasks(
+  resturl: string,
+  user_addr: string,
+  image_md5s: string[],
+  priv: string,
+  public_inputs: string,
+  private_inputs: string,
+  num_prove_tasks : number,
+  interval_ms : number,
+  total_time_ms : number,
+) {
+  sendIntervaledRequests(total_time_ms, interval_ms, num_prove_tasks,
+    async (i : number) => {
+      await addProvingTask(
+        resturl,
+        user_addr,
+        image_md5s[i % image_md5s.length],
+        public_inputs,
+        private_inputs,
+        priv
+      );
+    });
+}
+
+async function runQueryTasks(
+  resturl: string,
+  task_ids: string[],
+  num_query_tasks : number,
+  interval_ms : number,
+  total_time_ms : number,
+) {
+  sendIntervaledRequests(total_time_ms, interval_ms, num_query_tasks,
+    async (i : number) => {
+      await queryTask(
+        task_ids[i % task_ids.length],
+        resturl,
+        false,
+      );
+    });
+}
+
+export async function pressureTest(
+  resturl: string,
+  user_addr: string,
+  priv: string,
+  public_inputs: string,
+  private_inputs: string,
+  num_prove_tasks : number,
+  interval_prove_tasks_ms : number,
+  num_query_tasks : number,
+  interval_query_tasks_ms : number,
+  total_time_sec : number,
+) {
+
+  const images_detail = await getAvailableImages(resturl, user_addr);
```

Comment:
Suggest allow tester to input the md5 of the image for doing pressure test.

This will allow tester to test specific scenario (like specify and test multi images) by running multiple instances of the pressure tests